File: franchise_erp/public/js/serial_no.js
frappe.ui.form.on("Serial No", {

    refresh(frm) {

        if (frm.doc.item_code && !frm.doc.custom_style) {

            frappe.db.get_value(
                "Item",
                frm.doc.item_code,
                "custom_barcode_code"
            ).then((r) => {

                if (r.message) {

                    frm.set_value(
                        "custom_style",
                        r.message.custom_barcode_code
                    );

                    frm.save();
                }
            });
        }
    },

    item_code(frm) {

        if (frm.doc.item_code) {

            frappe.db.get_value(
                "Item",
                frm.doc.item_code,
                "custom_barcode_code"
            ).then((r) => {

                if (r.message) {

                    frm.set_value(
                        "custom_style",
                        r.message.custom_barcode_code
                    );
                }
            });
        }
    }
});

File: franchise_erp/custom/serial_no.py
import frappe


def update_serial_custom_style():

    serials = frappe.db.sql("""
        SELECT
            sn.name,
            sn.item_code,
            i.custom_barcode_code
        FROM `tabSerial No` sn
        LEFT JOIN `tabItem` i
            ON i.name = sn.item_code
        WHERE IFNULL(sn.custom_style, '') = ''
        AND IFNULL(i.custom_barcode_code, '') != ''
        ORDER BY sn.creation DESC
        LIMIT 500
    """, as_dict=1)

    for d in serials:

        frappe.db.set_value(
            "Serial No",
            d.name,
            "custom_style",
            d.custom_barcode_code,
            update_modified=False
        )

    frappe.db.commit()